### PR TITLE
Docs: restructure tree and unify pages

### PR DESCRIPTION
**Pull request by ozkatz**
Trying to simplify the flow of getting a working lakeFS installation (non-local)
Additionally, removed the standalone "spark" page which didn't provide any useful information on its own.

This is part of wider reorganization of the documentation structure.

### CODE REVIEW

**Review by johnnyaug — CHANGES_REQUESTED** [review]
Thanks! 
We are now missing the part of creating the first user and the first repository.

File: docs/deploy/onprem.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
+---
+layout: default
+title: On-Prem
+parent: Deploy and Setup lakeFS
+description: This section will guide you through deploying and setting up a production-suitable lakeFS environment on premise
+nav_order: 40
```
[the remark this answers is not in the document]
Comment (ozkatz) [reply docs/deploy/onprem.md]: Done

File: docs/deploy/onprem.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+     s3:
+        force_path_style: true
+        endpoint: http://<minio_endpoint>
+        discover_bucket_region: false
+        credentials:
+           access_key_id: <minio_access_key>
+           secret_access_key: <minio_secret_key>
+   ```
+
+   ⚠️ Notice that the lakeFS Blockstore type is set to `s3` - This configuration works with S3-compatible storage engines such as [MinIO](https://min.io/){: target="blank" }.
```
[the remark this answers is not in the document]
Comment (ozkatz) [reply docs/deploy/onprem.md:72]: True! Added as prerequisite

File: docs/deploy/gcp.md [hunk 3]
```diff
@@ -1,38 +1,48 @@
...
-description: This guide will help you deploy your production lakeFS environment on Google Cloud Platform (GCP).
-nav_order: 30
-next: ["Prepare your storage", "../setup/storage/index.html"]
+title: GCP
+parent: Deploy and Setup lakeFS
+description: This section will guide you through deploying and setting up a production-suitable lakeFS environment on Google Cloud Platform (GCP).
+nav_order: 50
+redirect_from:
+   - ../setup/storage/gcs.html 
+next:  ["Import data into your installation", "../howto/import.html"]
```
[the remark this answers is not in the document]
Comment (ozkatz) [reply docs/deploy/gcp.md:9]: Agree! Added to all deployment guides these missing steps.

File: docs/howto/index.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+layout: default
+title: How-To
+description: 
+nav_order: 60
```
[the remark this answers is not in the document]
Comment (ozkatz) [reply docs/howto/index.md:5]: Hopefully will very soon resolve itself with the new hierarchy and order :)

[the remark this answers is not in the document]
Comment (ozkatz) [reply docs/howto/index.md:5]: Done!

**Review by johnnyaug — COMMENTED** [review]
Looks really, really good. 
There are still some comments - not all of them need to be handled in this PR, but I think some do.

**Comment by ozkatz** [comment]
@johnnyaug Would appreciate the help! I accept all comments you've made but haven't had the time to implement them. If you can pick up the rest I'll be happy to approve and we can merge today!

**Comment by johnnyaug** [comment]
@ozkatz on it

**Review by johnnyaug — APPROVED** [review]

**Pull request merged**